File: IKS/uber/src/app/service/statistics.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class StatisticsService {
  constructor(private http: HttpClient) {}

  private getAuthHeaders() {
    const token = localStorage.getItem('auth_token');
    return {
      headers: new HttpHeaders({
        Authorization: `Bearer ${token}`,
      }),
    };
  }

  getMyStatistics(startDate: string, endDate: string): Observable<any> {
    return this.http.get(`${environment.apiHost}/statistics/my`, {
      params: { startDate, endDate },
      ...this.getAuthHeaders(),
    });
  }

  getAllUsersStatistics(
    startDate: string,
    endDate: string,
    userType: 'drivers' | 'passengers',
  ): Observable<any> {
    return this.http.get(`${environment.apiHost}/statistics/all/${userType}`, {
      params: { startDate, endDate },
      ...this.getAuthHeaders(),
    });
  }

  getUserStatistics(userId: number, startDate: string, endDate: string): Observable<any> {
    return this.http.get(`${environment.apiHost}/statistics/user/${userId}`, {
      params: { startDate, endDate },
      ...this.getAuthHeaders(),
    });
  }

  getUsers(userType: 'drivers' | 'passengers'): Observable<any[]> {
    return this.http.get<any[]>(
      `${environment.apiHost}/statistics/users/${userType}`,
      this.getAuthHeaders(),
    );
  }
}

File: MOBILNE/app/src/main/java/com/example/ubercorp/api/StatisticsService.java
package com.example.ubercorp.api;

import com.example.ubercorp.dto.StatisticsDTO;
import com.example.ubercorp.dto.UserBasicDTO;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface StatisticsService {
    @GET("api/statistics/my")
    Call<StatisticsDTO> getMyStatistics(
            @Header("Authorization") String token,
            @Query("startDate") String startDate,
            @Query("endDate") String endDate
    );

    @GET("api/statistics/all/{userType}")
    Call<StatisticsDTO> getAllUsersStatistics(
            @Header("Authorization") String token,
            @Path("userType") String userType,
            @Query("startDate") String startDate,
            @Query("endDate") String endDate
    );

    @GET("api/statistics/user/{userId}")
    Call<StatisticsDTO> getUserStatistics(
            @Header("Authorization") String token,
            @Path("userId") Long userId,
            @Query("startDate") String startDate,
            @Query("endDate") String endDate
    );

    @GET("api/statistics/users/drivers")
    Call<List<UserBasicDTO>> getAllDrivers(@Header("Authorization") String token);

    @GET("api/statistics/users/passengers")
    Call<List<UserBasicDTO>> getAllPassengers(@Header("Authorization") String token);
}
